pkg/agent/agent_linux.go

```diff
@@ -274,36 +261,33 @@ func (i *Initializer) RestoreOVSBridge() {
 	}
 	klog.Infof("Restoring bridge config to uplink...")
 	uplinkNetConfig := i.nodeConfig.UplinkNetConfig
-	uplink := ""
+	uplinkName := ""
+	bridgedUplinkName := ""
 	if uplinkNetConfig != nil {
-		uplink = uplinkNetConfig.Name
+		uplinkName = uplinkNetConfig.Name
+		bridgedUplinkName = util.GetBridgedUplinkName(uplinkName)
 	}
+	brName := i.ovsBridge
 
-	brName := i.ovsBridgeClient.GetBridgeName()
-	brLink, err := netlink.LinkByName(brName)
+	uplinkIPs, err := getTransportIPNetDeviceByNameWithExcludedList(uplinkName, nil)
 	if err != nil {
-		klog.Warningf("OVS bridge local port not found: %+v", err)
+		klog.ErrorS(err, "Failed to get uplink IPs")
 	}
-
-	if uplink != "" {
-		if err := util.DeleteOVSPort(brName, uplink); err != nil {
-			klog.ErrorS(err, "Removing uplink port from bridge failed", "uplink", uplink, "bridge", brName)
+	if uplinkName != "" {
+		if err := util.DeleteOVSPort(brName, uplinkName); err != nil {
+			klog.ErrorS(err, "Delete OVS port failed", "port", uplinkName)
 		}
-		if err := util.ConfigureLinkAddresses(uplinkNetConfig.Index, []*net.IPNet{uplinkNetConfig.IP}); err != nil {
-			klog.ErrorS(err, "Configure IP to uplink failed", "uplink", uplink)
+		if err := util.DeleteOVSPort(brName, bridgedUplinkName); err != nil {
+			klog.ErrorS(err, "Delete OVS port failed", "port", bridgedUplinkName)
 		}
-	}
-	if brLink != nil {
-		if err := util.ConfigureLinkAddresses(brLink.Attrs().Index, nil); err != nil {
-			klog.ErrorS(err, "Remove IP from bridge interface failed", "interface", brName)
+		if err := util.ChangeAdapterName(bridgedUplinkName, uplinkName, true); err != nil {
```

Comment:
Add a check for the existence of the netlink with uplink name before rename uplink back, otherwise it may get an error that the name is already existing on the host.

Reply:
Yes, but if error occurred here, there must be another error occurred before,  thus the error log can report the unexpected status here.
Meanwhile, all errors will be ignored in this func to restore the network status.

Reply:
Another candidate issue I want to mention is, we should ensure the uplink interface is removed from OVS datapath ( ovs-dpctl show) completely when restoring the configuration. Otherwise, the Node network may have connectivity issues although the L3 configurations are moved back. This was found in VM case, but I think it should also exist with flexible IPAM scenario. I doubt the issue was because the uplink interface is busy when OVS tries to remove it in datapath, and OVS doesn't retry later.

Reply:
I think the major difference is that FlexibleIPAM uses ovs-vsctl to delete uplink port when graceful shutdown,  and we never hit the issue.

Reply:
As I mentioned above, the delete port method is different.
I'll test more for graceful shutdown to check if this issue will occur.